topology/src/main/java/io/camunda/zeebe/topology/serializer/ProtoBufSerializer.java

```diff
@@ -264,11 +267,20 @@ private TopologyChangeOperation decodeOperation(
   }
 
   @Override
-  public byte[] encode(final AddMembersRequest addMembersRequest) {
-    return Requests.AddMemberRequest.newBuilder()
-        .addAllMemberIds(addMembersRequest.members().stream().map(MemberId::id).toList())
-        .build()
-        .toByteArray();
+  public byte[] encodeRequest(final TopologyManagementRequest topologyManagementRequest) {
```

Comment:
:wrench: Can also define separate `encode` method for each request type so that it is statically bound. I see no advantage in resolving the type during runtime.
```
byte[] encode(AddMembersRequest);
byte[] encode(JoinPartitionRequest);
```